src/field-selection.ts

```diff
@@ -52,13 +52,19 @@ function expandPath(object: any, path: string | string[], currPath: readonly str
   const rest = segments.slice(1);
 
   if (object != null && !_.isObjectLike(object)) {
-    if (key === '**' && !rest.length) {
-      // globstar leaves are always selected
-      return [reconstructFieldPath(currPath)];
+    if (key === '**') {
+      if (!rest.length) {
+        // globstar leaves are always selected
+        return [reconstructFieldPath(currPath)];
+      }
+      return [];
     }
-
-    // there still are paths to traverse, but value is a primitive, stop
-    return [];
+    if (key === '*') {
+      // wildcard position does not exist
+      return [];
+    }
+    // value is a primitive, there are still still paths to traverse that will be likely undefined, we return the entire path
```

Comment:
Fun fact: you could theoretically go forever now (not for any good reason though), e.g. `foo.toFixed.name.toString.length.toFixed`... you just have to keep diving deeper into the prototype!

```suggestion
    // value is a primitive, paths being traversed from here might be in their prototype, return the entire path
```